lib/logstash/filters/mutate.rb

```diff
@@ -228,6 +240,13 @@ def remove(event)
   end # def remove
 
   private
+  def keep(event)
+    event.to_hash.each do |name, value|
+      event.remove(name) unless (@keep.include?(name) || name[0] == '@')
```

Comment:
we might want to support `sprintf` in the `keep` field names here. for example `error_%{code}`. since previous filters may create such fields dynamically.

also, I believe the _only_ reserved `@` fields are `@timestamp` and `@version`. maybe we can figure out a way to explicitly include them here, and allow other `@*` fields to be allowed to be removed from Event.